src/zenml/stack/flavor.py

```diff
@@ -139,20 +148,37 @@ def to_model(
         )
         return model
 
-    def generate_default_docs_url(self) -> str:
+    def generate_default_docs_url(self, component_name: str = None) -> str:
         """Generate the doc urls for all inbuilt and integration flavors.
 
         Note that this method is not going to be useful for custom flavors,
         which do not have any docs in the main zenml docs.
 
+        Args:
+            component_name: The name of the component for docs generation. Used
+                for legacy documentation before ZenML v0.34.0.
+
         Returns:
-            The complete url to the zenml docs
+            The complete url to the zenml documentation
         """
         from zenml import __version__
 
         component_type = self.type.plural.replace("_", "-")
         name = self.name.replace("_", "-")
+        docs_component_name = component_name or name
         base = f"https://docs.zenml.io/v/{__version__}"
-        url = f"{base}/component-gallery/{component_type}/{name}"
+        return (
+            f"{base}/component-gallery/{component_type}/{docs_component_name}"
```

Comment:
Isnt the URL directly generated from the file name rather than the `component_type` and `docs_component_name`? However, I do see the problem that the files are not accessible to the package... Therefore, we must make sure the filename conform to this

Reply:
I think #1187 addressed most of this renaming. I don't think there was anything that happened that would affect the API docs in this way, however.

Reply:
ok